### PR TITLE
Feat(projects.page.jsx)/ build project page

### PR DESCRIPTION
**Pull request by NabilSli**

### CODE REVIEW

File: my-portfolio/src/components/cards.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export const Card = ({ title, description, id, icon }) => {
+  return (
+    <Link to={`/projects/${id}`}>
+      <div className="cardContainer">
+        {icon && <img className="cardIcon" alt="icon" src={icon}></img>}
+        {title && <h2 className="cardTitle">{title}</h2>}
+        {description && <p className="cardDescription">{description}</p>}
```
Comment (tony-go) [inline my-portfolio/src/components/cards.jsx:10]: A good practice is to have a fallback of each of theses:

```suggestion
        <img className="cardIcon" alt="icon" src={icon || fallbackIcon }></img>
        <h2 className="cardTitle">{title || fallbackTitle}</h2>
        <p className="cardDescription">{description || fallbackDescription}</p>
```

File: my-portfolio/src/components/cards.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export const Card = ({ title, description, id, icon }) => {
```
Comment (tony-go) [inline my-portfolio/src/components/cards.jsx:4]:
```suggestion
export const Card = ({ title, description, id, iconSrc }) => {
```

File: my-portfolio/src/components/displayPictures.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const DisplayPictures = ({ pictures }) => {
+  return (
+    <div className="pictureContainer">
+      <a>
+        <img
+          className="projectPictures"
+          src={pictures}
+          alt="image du projet presenter"
```
Comment (tony-go) [inline my-portfolio/src/components/displayPictures.jsx:8]:
```suggestion
          alt="image du projet presenté"
```

File: my-portfolio/src/components/pdfViewer.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+const PdfViewer = () => {
+  return (
+    <div className="pdfViewer">
+      <iframe src="\cv.pdf" width="60%" height="1200px" />
```
Comment (tony-go) [inline my-portfolio/src/components/pdfViewer.jsx:6]: `"\cv.pdf"` I concerned about `\` why did you use it? Windows?

File: my-portfolio/src/pages/Projects.page.jsx [hunk 5]
```diff
@@ -1,9 +1,50 @@
-import React from "react";
+import React, { useState } from "react";
+
+import { useParams, useNavigate } from "react-router-dom";
+
+import fetchData from "../components/fetchData";
```
Comment (tony-go) [inline my-portfolio/src/pages/Projects.page.jsx:5]:
```suggestion
import useProjects from "../components/fetchData";
```

File: my-portfolio/src/pages/Projects.page.jsx [hunk 6]
```diff
@@ -1,9 +1,50 @@
...
+    return navigate("/404");
+  }
+
+  const currentProject = projects.find((project) => project.id === id);
+  if (!currentProject || currentProject.length <= 0) {
+    return navigate("/404");
+  }
+  const title = currentProject.title;
+  const pictures = currentProject.pictures;
+  const description = currentProject.description;
```
Comment (tony-go) [inline my-portfolio/src/pages/Projects.page.jsx:31]:
```suggestion
  const { title, pictures, description } = currentProject;
```

File: my-portfolio/src/components/pdfViewer.jsx (hunk 4 above)
Comment (NabilSli) [reply my-portfolio/src/components/pdfViewer.jsx:6]: it's because a copied the path ... i fixed it

**Review by tony-go — APPROVED** [review]

**Pull request merged**